### PR TITLE
Add bond e2e test

### PR DESCRIPTION
**Pull request by qinqon**
The test case is taken from cnv-test use of bond interfaces

Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

File: test/e2e/nodenetworkstate_test.go [hunk 1]
```diff
@@ -30,8 +42,41 @@ var _ = Describe("NodeNetworkState", func() {
...
+        stp:
+          enabled: false
+      port:
+        - name: bond1
+          stp-hairpin-mode: false
+          stp-path-cost: 100
+          stp-priority: 32
+`)
 
+		br1Absent = nmstatev1alpha1.State(`interfaces:
```
Comment (phoracek) [inline test/e2e/nodenetworkstate_test.go]: what about absent bonds? do we cleanup connections on host (is there leftover of deactivated connection)?

Comment (qinqon) [reply test/e2e/nodenetworkstate_test.go]: I leave it on purpose, as it does exactly same thing as bridges just link del.

Comment (phoracek) [reply test/e2e/nodenetworkstate_test.go]: i don't understand.

why do we have it for bridge and not for bond?

do we have leftover connections on node? if so, it affects the test. in real situation, we would have no pre-existing connection. but in this test, we reuse the same over and over again (wild guess)

File: test/e2e/nodenetworkstate_test.go [hunk 2]
```diff
@@ -75,10 +116,81 @@ var _ = Describe("NodeNetworkState", func() {
...
+							HaveKeyWithValue("type", expectedBond["type"]),
+							HaveKeyWithValue("state", expectedBond["state"]),
+							HaveKeyWithValue("link-aggregation", expectedBond["link-aggregation"]),
+						)),
+						ContainElement(SatisfyAll(
+							HaveKeyWithValue("name", expectedBridge["name"]),
+							HaveKeyWithValue("type", expectedBridge["type"]),
+							HaveKeyWithValue("state", expectedBridge["state"]),
+							HaveKeyWithValue("bridge", HaveKeyWithValue("port", expectedBridgePorts)),
+						))))
```
Comment (phoracek) [inline test/e2e/nodenetworkstate_test.go]: lisp vibes

**Pull request merged**